certbot/main.py

```diff
@@ -860,6 +862,37 @@ def plugins_cmd(config, plugins):
     logger.debug("Prepared plugins: %s", available)
     notify(str(available))
 
+def enhance(config, plugins):
+    """Add security enhancements to existing configuration
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param plugins: List of plugins
+    :type plugins: `list` of `str`
+
+    :returns: `None`
+    :rtype: None
+
+    """
+    supported_enhancements = ["hsts", "redirect", "staple", "uir"]
+    # Check that at least one enhancement was requested on command line
+    if not any([getattr(config, enh) for enh in supported_enhancements]):
+        msg = ("Please specify one or more enhancement types to configure. To list "
+               "the available enhancement types, run:\n\n%s --help enhance\n")
+        logger.warning(msg, sys.argv[0])
+        raise errors.MisconfigurationError("No enhancements requested, exiting.")
+
+    try:
+        installer, _ = plug_sel.choose_configurator_plugins(config, plugins, "enhance")
+    except errors.PluginSelectionError as e:
+        return str(e)
+    le_client = _init_le_client(config, authenticator=None, installer=installer)
+    domain_question = ("Which domain names would you like to enable the selected "
+                       "enhancements for")
+    domains, _ = _find_domains_or_certname(config, installer, domain_question)
+    le_client.enhance_config(domains, None, ask_redirect=False)
```

Comment:
`certbot enable --nginx --staple-ocsp` won't work without the chain path. I think this should be fixed before we ship this. I'd even be fine with us being unable to modify configs we're not managing a cert for, at least for now.

If this makes it significantly harder to support `--domains`, I'd suggest not doing this now. We can always do it later when we have spare cycles and/or there's a demand for it.

Reply:
We might want to skip it for now. I'll enable it when I'll be working on the prefetch-OCSP functionality.

Reply:
I asked @ohemorange about this because I think she has a stronger opinion here than I do and she has a strong preference for us only taking `--cert-name`. Her main argument was we should limit the number of ways there are to accomplish a task in Certbot. This makes both the code and documentation simpler.

If we take `--domains`, we'll need a way to get the chain path in the future. With the way this code is now, if you request `--staple-ocsp`, it will error and say that `--chain-path` needs to be provided, but even if you provide it, we'll still error because it's not being passed through to `enhance_config`. We could fix that by providing the value of `chain_path` from the `config`, but we might instead want to try and find the matching certificate for the given domains. Doing this opens a massive cans of worms which I don't think we want.

I don't have that strong of an opinion, so you and Erica can debate it if you want, but I'll say my preference would be to err on the side of providing fewer bells and whistles initially by not supporting `--domains` since support can always be added later if there is significant demand/need for the feature.